app/models/custom_filter.rb

```diff
@@ -21,4 +21,11 @@ class CustomFilter < ApplicationRecord
   belongs_to :account
 
   enum filter_type: { conversation: 0, contact: 1, report: 2 }
+  validate :validate_number_of_filters
+
+  def validate_number_of_filters
+    return true if self.class.where(account_id: account_id, user_id: user_id).size <= 50
```

Comment:
account.custom_filters.where(user_id: user_id).size ?

Reply:
also this could be a constant `MAX_FILTER_PER_USER` . so that you can mock it in the spec to a lower value  like 2 or 3. will reduce the complexity of creating 51 objects for the spec